src/pages/iou/IOUTransactions.js

```diff
@@ -18,65 +17,36 @@ const propTypes = {
 
     /** ReportID for the associated IOU report */
     iouReportID: PropTypes.number.isRequired,
-
-    /** Transactions for this IOU report */
-    transactions: PropTypes.arrayOf(PropTypes.shape(iouTransactionPropTypes)),
 };
 
 const defaultProps = {
     reportActions: {},
-    transactions: [],
 };
 
-class IOUTransactions extends Component {
-    constructor(props) {
-        super(props);
-
-        this.getActionForTransaction = this.getActionForTransaction.bind(this);
-    }
-
-    /**
-     * Given a transaction from an IOU Report, returns the chatReport action with a matching transactionID. Unless
-     * something has gone wrong with our storing logic, there should always exist an action for each transaction.
-     *
-     * @param {Object} transaction
-     * @returns {Object} action
-     */
-    getActionForTransaction(transaction) {
-        const matchedAction = _.find(this.props.reportActions, (action) => {
-            // iouReport.transaction.transactionID is returned as a String, but the originalMessage value is Number
-            if (action && action.originalMessage && action.originalMessage.IOUTransactionID
-                && action.originalMessage.IOUTransactionID.toString() === transaction.transactionID) {
-                return action;
+const IOUTransactions = ({
+    reportActions,
+    chatReportID,
+    iouReportID,
+}) => (
+    <View style={[styles.mt3]}>
+        {_.map(reportActions, (reportAction) => {
+            if (reportAction.actionName === 'IOU'
+                && reportAction.originalMessage.IOUReportID === iouReportID) {
+                // Cancelled/declined actions reuse the original transactionID, so we must make the key unique
+                const transactionKey = `${reportAction.originalMessage.IOUTransactionID}-
```

Comment:
This is not ensured to be unique, can we use the reportAction's sequence number instead?

Reply:
Yep, much better. Done 👍